crucible/src/Lang/Crucible/Simulator/OverrideSim.hs

```diff
@@ -159,11 +162,14 @@ bindOverrideSim (Sim m) h = Sim $ unSim . h =<< m
 instance Monad (OverrideSim p sym ext rtp args r) where
   return = returnOverrideSim
   (>>=) = bindOverrideSim
-  fail msg = Sim $ StateContT $ \_c -> runGenericErrorHandler msg
```

Comment:
I don't know what the backward-compatibility story is here, but I don't want to force downstream clients to hide `Prelude.fail` and add `MonadFail` constraints just yet, so we should probably continue defining the `Monad.fail` method for now.

Should we use CPP to conditionally define this method on 8.6 and below?

Reply:
CPP sounds like a good way to do it.